packages/stratimikos/adapters/belos/src/tpetra/Thyra_BelosTpetraPreconditionerFactory_def.hpp

```diff
@@ -76,6 +76,7 @@
 
 namespace Thyra {
 
+using Teuchos::rcp;
```

Comment:
This needs to be removed and put into the individual function bodies where it is needed.  It is not safe to put a using declaration for a function in a header file like this.  See Appendix D.2 "Amendments for ’using’ declarations and directives" in [Thyra Coding and Documentation Guidelines (TCDG) Version 1.0](https://docs.trilinos.org/dev/packages/thyra/doc/html/ThyraCodingGuideLines.pdf) and  Item 59 "Don't write namespace usings in a header file before an #include" in ["C++ Coding Standards"](https://dl.acm.org/doi/abs/10.5555/1036281).